### PR TITLE
redpanda: convert NOTES.txt to go

### PR DESCRIPTION
**Pull request by jan-g**
I didn't add the Ci machinery to compare a whole raft of (pretty much identical) NOTES.txt output, just ran it locally.

The `_example-commands.tpl` script calls into the gotohelm functions now.

### CODE REVIEW

**Comment by RafalKorepta** [comment]
https://github.com/redpanda-data/helm-charts/actions/runs/9760792194/job/26940439905?pr=1403#step:5:359

I believe it's the difference of running `nix develop -c 'task ci:lint'` or `nix develop -c 'gofumpt -w .'`

```
task: [ci:lint] gofumpt -w .
task: [ci:lint] git diff --exit-code
diff --git a/charts/redpanda/notes.go b/charts/redpanda/notes.go
index 05d11b2..8099fe9 100644
--- a/charts/redpanda/notes.go
+++ b/charts/redpanda/notes.go
@@ -149,7 +149,6 @@ func Notes(dot *helmette.Dot) []string {
 			``,
 			fmt.Sprintf(`  %s`, RpkACLCreate(dot)),
 		)
-
 	}
 	notes = append(notes,
 		``,
task: Failed to run task "ci:lint": exit status 1
```

**Comment by chrisseto** [comment]
> I believe it's the difference of running nix develop -c 'task ci:lint' or nix develop -c 'gofumpt -w .'

Huh... Those should be the same 🤔

**Comment by RafalKorepta** [comment]
I found small problem
```
2024-07-02T14:09:16.2034506Z >>> helm install redpanda-pfzd2eoqxt charts/redpanda --namespace redpanda-pfzd2eoqxt --wait --values charts/redpanda/ci/96-audit-logging-values.yaml --timeout 600s --set=image.tag=
2024-07-02T14:09:16.3989941Z Error: INSTALLATION FAILED: YAML parse error on redpanda/templates/tests/test-kafka-sasl-status.yaml: error converting YAML to JSON: yaml: line 45: could not find expected ':'
```

https://github.com/redpanda-data/helm-charts/actions/runs/9762101576/job/26944710395?pr=1403#step:16:6299

**Comment by RafalKorepta** [comment]
It could be reproduced by 
```
helm template charts/redpanda --namespace redpanda-pfzd2eoqxt --values charts/redpanda/ci/96-audit-logging-values.yaml.tpl
Error: YAML parse error on redpanda/templates/tests/test-kafka-sasl-status.yaml: error converting YAML to JSON: yaml: line 45: could not find expected ':'
```

File: charts/redpanda/templates/_example-commands.tpl [hunk 1]
```diff
@@ -23,30 +23,36 @@ and tested in a test.
 
 {{/* tested in tests/test-kafka-sasl-status.yaml */}}
 {{- define "rpk-acl-user-create" -}}
-rpk acl user create myuser --new-password changeme --mechanism {{ include "sasl-mechanism" . }}
+{{- $cmd := (get ((include "redpanda.RpkACLUserCreate" (dict "a" (list .))) | fromJson) "r") }}
+{{ $cmd }}
```
[the remark this answers is not in the document]
Comment (jan-g) [reply charts/redpanda/templates/_example-commands.tpl:27]: Ah-ha, thanks.

**Review by RafalKorepta — APPROVED** [review]
I don't understand why imports should be changed. Never the less LGTM

https://github.com/redpanda-data/helm-charts/actions/runs/9775423743/job/26985777471?pr=1403#step:5:361

```
+++ b/charts/redpanda/values_partial.gen.go
@@ -6,7 +6,7 @@
 package redpanda
 
 import (
-	cmmetav1 "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
+	cmmeta "github.com/cert-manager/cert-manager/pkg/apis/meta/v1"
```

**Comment by jan-g** [comment]
I'm not sure about the import thing either. It seems to come and go as I run the code generation; I can't fathom why.

**Pull request merged**